src/commands/join.cpp

```diff
@@ -22,103 +23,117 @@
 #include "../includes/utils.hpp"
 
 void Server::joinChannel(const std::string &param, int fd) {
-  // Check if the user is attempting to leave all channels (using "0" as param)
+  Client &client = _clients.at(fd);
+
+  if (param.empty() || (param.length() == 1 && param[0] == REG_CHAN)) {
+    send461NeedMoreParams(client, "JOIN");
+    return;
+  }
   if (isLeaveAllChannelsRequest(param)) {
-    // partAllChannels(fd); // Leave all channels function
+    quitChannel(fd);
     return;
   }
-
-  Client &client = _clients.at(fd);
+  stringVector channels;
   std::string::size_type spacePos = param.find(" ");
   std::string channelsPart = param.substr(0, spacePos);
```

Comment:
[Parsing issue]
For the command `JOIN #42, #43`
--> channelsPart = `#42,`. `#43` is missing.